termux-exec.c

```diff
@@ -230,9 +233,28 @@ int execve(const char* filename, char* const* argv, char* const* envp)
 			}
 		}
 	}
+	*filename_ptr = (char*)filename;
 
-	int ret = real_execve(filename, argv, envp);
 	free(new_argv);
 	free(new_envp);
+}
+
+
+int execve(const char* filename, char* const* argv, char* const* envp)
+{
+	int (*real_execve)(const char*, char* const[], char* const[]) = dlsym(RTLD_NEXT, "execve");
+	update_filename_argv_envp(&filename, &argv, &envp);
+	int ret = real_execve(filename, argv, envp);
 	return ret;
 }
+
+
+int execvp(const char* filename, char* const argv[])
+{
+	char* const* envp = NULL;
+	int (*real_execvp) (const char*, char* const[]) = dlsym(RTLD_NEXT, "execvp");
+	update_filename_argv_envp(&filename, &argv, &envp);
```

Comment:
Are you sure `envp` can be NULL for this function? I'm worried about NULL-ptr-deref bugs here.